packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -333,9 +335,43 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
         return editor;
     }
 
+    private readonly updateLinkNodes = (nodeArray: NodeObj[]): void => {
+        nodeArray.forEach((node: NodeObj) => {
+            if (node.type !== 'text' && node.content) {
+                this.updateLinkNodes(node.content);
+            } else {
+                node.marks?.forEach((mark, index: number) => {
+                    if (mark.type === 'link' && mark.attrs) {
+                        if (this.validAbsoluteLinkRegex.test(mark.attrs.href)) {
```

Comment:
Why do we need to manually do this validation here? Shouldn't the Link mark validation be handling this for us downstream?

Reply:
Actually, the validation added to the Tiptap editor in the Link mark is not working for the pasted content containing links. The validation is added only for the input rule i.e. by typing into the editor and pressing any special keys to convert it as a link. We found that there is an open issue in Tiptap explaining that the validation is not added for the pasted links. 

I captured this here - https://github.com/ni/nimble/issues/1527 and once the underlying Tiptap issue is resolved, we can remove this manual validation from here. 
![image](https://github.com/ni/nimble/assets/123377523/9384f4d3-fa0b-452c-9180-49e01491f7ed)

I also saw the discussion in the above linked Tiptap where one mentioned it is resolved with the latest version `2.1.11` of Tiptap but when I locally bumped to the latest version, I could still reproduce the issue. And also noticed there are some other issues with the `linkOnPaste` behavior with the latest version so dropped the plan of updating the version as of now.